v4/core/base_service.go

```diff
@@ -451,3 +480,157 @@ func getErrorMessage(responseMap map[string]interface{}, statusCode int) string
 	// for the status code.
 	return http.StatusText(statusCode)
 }
+
+// EnableRetries will construct a "retryable" HTTP Client with the specified
+// configuration, and then set it on the service instance.
+// If maxRetries and/or maxRetryInterval are specified as 0, then default values
+// are used instead.
+func (service *BaseService) EnableRetries(maxRetries int, maxRetryInterval time.Duration) {
+	client := NewRetryableHTTPClient()
+	if maxRetries > 0 {
+		client.RetryMax = maxRetries
+	}
+	if maxRetryInterval > 0 {
+		client.RetryWaitMax = maxRetryInterval
+	}
+
+	service.SetHTTPClient(client.StandardClient())
+}
+
+// DisableRetries will disable automatic retries by constructing a new
+// default (non-retryable) HTTP Client instance and setting it on the service.
+func (service *BaseService) DisableRetries() {
+	service.SetHTTPClient(DefaultHTTPClient())
+}
+
+// DefaultHTTPClient returns a non-retryable http client with default configuration.
+func DefaultHTTPClient() *http.Client {
+	return cleanhttp.DefaultPooledClient()
+}
+
+// httpLogger is a shim layer used to allow the Go core's logger to be used with the retryablehttp interfaces.
+type httpLogger struct {
+}
+
+func (l *httpLogger) Printf(format string, inserts ...interface{}) {
+	GetLogger().Log(LevelDebug, format, inserts...)
+}
+
+// NewRetryableHTTPClient returns a new instance of go-retryablehttp.Client
+// with a default configuration that supports Go SDK usage.
+func NewRetryableHTTPClient() *retryablehttp.Client {
+	client := retryablehttp.NewClient()
+	client.Logger = &httpLogger{}
```

Comment:
We'll use a "logger" that logs messages at "debug" level with the retryablehttp function.   This means that if the default go core logger (error level) is in use, no messages from the retryablehttp package will be logged.   If the go core logger is logging at "debug" level, then we'll see the messages.